integrations/ruby/executor.rb

```diff
@@ -0,0 +1,135 @@
+require 'json'
+require 'mongo'
+
+Mongo::Logger.logger.level = Logger::WARN
+
+class UnknownOperation < StandardError; end
+class UnknownOperationConfiguration < StandardError; end
+
+class Executor
+  def initialize(uri, spec)
+    @uri, @spec = uri, spec
+    @operation_count = @failure_count = @error_count = 0
+  end
+
+  attr_reader :uri, :spec
+  attr_reader :operation_count, :failure_count, :error_count
+
+  def run
+    set_signal_handler
+    load_data
+    while true
+      break if @stop
+      perform_operations
+    end
+    puts "Result: #{result.inspect}"
+    write_result
+  end
+
+  private
+
+  def set_signal_handler
+    Signal.trap('INT') do
+      @stop = true
+    end
+  end
+
+  def load_data
```

Comment:
The testData key has to be ignored by the workload executor - see the `testData` key documentation at https://mongodb-labs.github.io/drivers-atlas-testing/spec-test-format.html - I am not sure why you are first deleting and then re-inserting the testData into the cluster here. Is it to make it easier to run locally or to make the script self-sufficient?

Reply:
I removed the call to this method, with a note to reinstate it when running workload executor by itself.

Reply:
Great!